testhelper/integrationtesthelper.go

```diff
@@ -35,29 +44,115 @@ const (
+		expectedPos = r3.Vector{X: 7.507596391989648, Y: 3.193198802065579, Z: 0}
+		expectedOri = &spatialmath.R4AA{
+			RX:    0,
+			RY:    0,
+			RZ:    1,
+			Theta: 0.001955831550003536,
+		}
+
+	case runtime.GOOS == "darwin" && useIMU:
+		expectedPos = r3.Vector{X: 1.6456359928659154, Y: 7.359399690067484, Z: 0}
+		expectedOri = &spatialmath.R4AA{
+			RX:    0.9862302383600338,
+			RY:    0.1635349032143911,
+			RZ:    -0.02462219273278937,
+			Theta: 0.025088490446011937,
+		}
+	case runtime.GOOS == "linux" && useIMU:
+		expectedPos = r3.Vector{X: 5.360365602699378, Y: 4.487447113637527, Z: 0}
+		expectedOri = &spatialmath.R4AA{
+			RX:    0.9792752063682609,
+			RY:    0.16419980776934567,
+			RZ:    0.11856851741044075,
+			Theta: 0.02525907400623705,
+		}
+	}
+
+	position, componentRef, err := svc.Position(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, componentRef, test.ShouldEqual, expectedComponentRef)
+
+	pos := position.Point()
+	t.Logf("Position point: (%v, %v, %v)", pos.X, pos.Y, pos.Z)
+	test.That(t, pos.X, test.ShouldAlmostEqual, expectedPos.X, tolerancePos)
+	test.That(t, pos.Y, test.ShouldAlmostEqual, expectedPos.Y, tolerancePos)
+	test.That(t, pos.Z, test.ShouldAlmostEqual, expectedPos.Z, tolerancePos)
+
+	ori := position.Orientation().AxisAngles()
+	t.Logf("Position orientation: RX: %v, RY: %v, RZ: %v, Theta: %v", ori.RX, ori.RY, ori.RZ, ori.Theta)
+	test.That(t, ori.RX, test.ShouldAlmostEqual, expectedOri.RX, toleranceOri)
+	test.That(t, ori.RY, test.ShouldAlmostEqual, expectedOri.RY, toleranceOri)
+	test.That(t, ori.RZ, test.ShouldAlmostEqual, expectedOri.RZ, toleranceOri)
+	test.That(t, ori.Theta, test.ShouldAlmostEqual, expectedOri.Theta, toleranceOri)
+}
+
+// Checks the cartographer map and confirms there at least 100 map points.
+func testCartographerMap(t *testing.T, svc slam.Service, localizationMode bool) {
+	timestamp1, err := svc.LatestMapInfo(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+	pcd, err := slam.PointCloudMapFull(context.Background(), svc)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, pcd, test.ShouldNotBeNil)
+	timestamp2, err := svc.LatestMapInfo(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+
+	if localizationMode {
+		test.That(t, timestamp1, test.ShouldResemble, timestamp2)
+	} else {
+		test.That(t, timestamp2.After(timestamp1), test.ShouldBeTrue)
+	}
+
+	pointcloud, err := pointcloud.ReadPCD(bytes.NewReader(pcd))
+	test.That(t, err, test.ShouldBeNil)
+	t.Logf("Pointcloud points: %v", pointcloud.Size())
+	test.That(t, pointcloud.Size(), test.ShouldBeGreaterThanOrEqualTo, 100)
+}
+
+// timeTracker stores the current and next timestamps for both IMU and lidar. These are used to manually
 // set the timestamp of each set of data being sent to cartographer and ensure proper ordering between them.
 // This allows for consistent testing.
-type TimeTracker struct {
-	LidarTime     time.Time
-	NextLidarTime time.Time
+type timeTracker struct {
```

Comment:
I could make a few structs and functions private in this function by moving `IntegrationCartographer` (named initially `testHelperCartographer`) into this file.